Yoast/wordpress-seo
# PC-362 whitespace within quotation marks breaks content analysis

## Context


* Inside `SentenceTokenizer.js`, we have a check whether the previous token is a period when the current token is a quotation mark:

> `if ( this.isQuotation( token.src ) && previousToken.src !== "." ) {`

* However, it's possible that the quotation mark occurs at the beginning of a text/block, hence the `previousToken` would be undefined.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where a text starting with a double quotation mark followed by a space(s) broke the analysis.
* [shopify-seo] Fixes a bug where a text starting with a double quotation mark followed by a space(s) broke the analysis.

## Relevant technical choices:

* I added an extra check for previousToken in this line:
`if ( this.isQuotation( token.src ) && previousToken && previousToken.src !== "." ) {`
* The extra check is added since it's possible that previousToken returns undefined if there is no token preceding the current token.


## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

#### Test in English
* Install and activate Yoast SEO
* Set the site language to English
* Create a post
* Add a text with 300 words to the post
* Add a double quotation mark followed by a space at the beginning of your text
* Confirm that no error appears in the console: i.e. `TypeError: Cannot read properties of undefined (reading 'src')`
* Confirm that the text length is 300 words: 
   * `Text length: The text contains 300 words. Good job!` 
* Repeat the step above for other types of double quotation mark `“”〝〞〟‟„『』«»`
* Make sure that still no error is found
* Install and activate Yoast SEO Premium
* Go back to your previous post and confirm that no error is still not found in the console
* Repeat the steps above on different content types: posts/pages/taxonomies/custom post types/custom taxonomies
* Repeat the steps above on different editors:  (Block/Classic/Elementor/other)
* Repeat the steps above in Shopify

#### Test in Japanese
* Do the testing with the site language set to Japanese
* Run the testing step in "Test in English" but with this quotation mark list &#x300C; &#x300E; &#x3008; &#x3014; &#x3010; &#xFF5B; &#xFF3B;

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes #
